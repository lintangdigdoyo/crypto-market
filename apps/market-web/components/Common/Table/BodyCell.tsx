import classNames from "classnames";

interface BodyCellProps extends React.TdHTMLAttributes<HTMLTableCellElement> {
  className?: string;
  children: React.ReactNode;
}

const BodyCell = ({ className, children, ...restProps }: BodyCellProps) => {
  return (
    <td
      className={classNames("p-5 border-t border-slate-200", className)}
      {...restProps}
    >
      {children}
    </td>
  );
};

export default BodyCell;
