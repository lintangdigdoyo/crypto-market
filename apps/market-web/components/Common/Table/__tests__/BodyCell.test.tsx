import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import BodyCell from "../BodyCell";

describe("Table components", () => {
  it("should renders BodyCell correctly", () => {
    const props = {
      children: <>children</>,
    };
    const component = renderer.create(<BodyCell {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
