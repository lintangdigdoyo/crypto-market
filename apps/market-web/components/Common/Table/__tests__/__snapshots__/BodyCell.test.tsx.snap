// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table components should renders BodyCell correctly 1`] = `
<td
  className="p-5 border-t border-slate-200"
>
  children
</td>
`;
